src/map/utils/fishingutils.cpp

```diff
@@ -379,11 +379,11 @@ namespace fishingutils
         return (uint16)std::floor(difficulty * ((rod->fishAttack + bonusAdd) / 100)) * 20;
     }
 
-    uint16 CaculateHeal(bool legendary, uint8 difficulty, rod_t* rod)
+    uint16 CalculateHeal(bool legendary, uint8 difficulty, rod_t* rod)
     {
         uint16 attack = CalculateAttack(legendary, difficulty, rod);
 
-        return (uint16)std::floor((attack / 20) * (rod->fishRecovery / 100)) * 10;
+        return (uint16)std::floor((attack / 20) * ((float)rod->fishRecovery / 100)) * 10;
```

Comment:
You probably want to make all the terms that are doing division or multiplication into floats, so that nothing is getting truncated.

The attack portion of this will still be getting truncated.

https://godbolt.org/z/KEnxc1nnh

Reply:
Roger, now that i'm looking everywhere else standby as I change a few more things.

Reply:
Lies

Reply:
If you're going to do conversions between numeric types, please use `static_cast` like I showed in the godbolt example.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast

Reply:
> Lies

What? (Why comment this?)

Reply:
Pushed with static casts.

Reply:
```cpp
    uint16 CalculateHeal(bool legendary, uint8 difficulty, rod_t* rod)
    {
        auto attack = CalculateAttack(legendary, difficulty, rod);
        return std::floor<uint16>((static_cast<float>(attack) / 20.0f) * (static_cast<float>(rod->fishRecovery) / 100.0f)) * 10.0f;
    }
```

Reply:
👍

Reply:
> > Lies
> 
> What? (Why comment this?)

When that comment was done, the conversation was closed as resolved with a "Addressed" comment on your part, but it really wasn't.

Reply:
>When that comment was done, the conversation was closed as resolved with a "Addressed" comment on your part, but it really wasn't.

Then can you comment with what needed to be further addressed? I'm still very basically learning cpp, so instead of commenting something that was completely useless to anyone, you could of instead helped me understand the issue.